python-client/giskard/scanner/robustness/text_transformations.py

```diff
@@ -116,10 +117,11 @@ class TextGenderTransformation(TextTransformation):
     name = "Switch gender"
 
     def make_perturbation(self, x):
+        language = detect_langs(x)[0].lang
```

Comment:
This should be calculated once in dataset text metadata provider (search `TextMetadataProvider`), and than accessed at the dataset level with `dataset.column_meta["my_textual_column", "text"].language`.